src/app/3d/qgs3daxis.cpp

```diff
@@ -0,0 +1,643 @@
+      mTextCoord_Y = QVector3D( 0.0f, mCylinderLength + coneLength / 2.0, 0.0f );
+
+      rotation = QQuaternion::fromAxisAndAngle( QVector3D( 0.0f, 0.0f, 0.0f ), 0.0f );
+      color = Qt::green;
+      text = mText_Y;
+      textTransform = mTextTransform_Y;
+      break;
+
+    case Axis::Z:
+      mText_Z = new Qt3DExtras::QText2DEntity( );  // object initialization in two step:
+      mText_Z->setParent( mTwoDLabelSceneEntity ); // see https://bugreports.qt.io/browse/QTBUG-77139
+      connect( mText_Z, &Qt3DExtras::QText2DEntity::textChanged, this, [this]( const QString & text )
+      {
+        mText_Z->setWidth( mFontSize * text.length() );
+      } );
+      mTextTransform_Z = new Qt3DCore::QTransform();
+      mTextCoord_Z = QVector3D( 0.0f, 0.0f, mCylinderLength + coneLength / 2.0 );
+
+      rotation = QQuaternion::fromAxisAndAngle( QVector3D( 1.0f, 0.0f, 0.0f ), 90.0f );
+      color = Qt::blue;
+      text = mText_Z;
+      textTransform = mTextTransform_Z;
+      break;
+
+    default:
+      return;
+  }
+
+  // cylinder
+  Qt3DCore::QEntity *cylinder = new Qt3DCore::QEntity( mAxisRoot );
+  Qt3DExtras::QCylinderMesh *cylinderMesh = new Qt3DExtras::QCylinderMesh;
+  cylinderMesh->setRadius( cylinderRadius );
+  cylinderMesh->setLength( mCylinderLength );
+  cylinderMesh->setRings( 10 );
+  cylinderMesh->setSlices( 4 );
+  cylinder->addComponent( cylinderMesh );
+
+  Qt3DExtras::QPhongMaterial *cylinderMaterial = new Qt3DExtras::QPhongMaterial( cylinder );
+  cylinderMaterial->setAmbient( color );
+  cylinderMaterial->setShininess( 0 );
+  cylinder->addComponent( cylinderMaterial );
+
+  Qt3DCore::QTransform *cylinderTransform = new Qt3DCore::QTransform;
+  QMatrix4x4 transformMatrixCylinder;
+  transformMatrixCylinder.rotate( rotation );
+  transformMatrixCylinder.translate( QVector3D( 0.0f, mCylinderLength / 2.0f, 0.0f ) );
+  cylinderTransform->setMatrix( transformMatrixCylinder );
+  cylinder->addComponent( cylinderTransform );
+
+  // cone
+  Qt3DCore::QEntity *coneEntity = new Qt3DCore::QEntity( mAxisRoot );
+  Qt3DExtras::QConeMesh *coneMesh = new Qt3DExtras::QConeMesh;
+  coneMesh->setLength( coneLength );
+  coneMesh->setBottomRadius( coneBottomRadius );
+  coneMesh->setTopRadius( 0.0f );
+  coneMesh->setRings( 10 );
+  coneMesh->setSlices( 4 );
+  coneEntity->addComponent( coneMesh );
+
+  Qt3DExtras::QPhongMaterial *coneMaterial = new Qt3DExtras::QPhongMaterial( coneEntity );
+  coneMaterial->setAmbient( color );
+  coneMaterial->setShininess( 0 );
+  coneEntity->addComponent( coneMaterial );
+
+  Qt3DCore::QTransform *coneTransform = new Qt3DCore::QTransform;
+  QMatrix4x4 transformMatrixCone;
+  transformMatrixCone.rotate( rotation );
+  transformMatrixCone.translate( QVector3D( 0.0f, mCylinderLength, 0.0f ) );
+  coneTransform->setMatrix( transformMatrixCone );
+  coneEntity->addComponent( coneTransform );
+
+  // text
+  QFont f = QFont( "monospace", mFontSize ); // TODO: should use outlined font
```

Comment:
Yes because we may need to use outlined font to have labels with better readability but theses fonts are not installed by default. I can remove the todo it, this problem have not solution for now.